src/Npgsql/Internal/Converters/ArrayConverter.cs

```diff
@@ -585,20 +584,29 @@ protected override PgConverter<T> CreateConverter(PgConverterResolution effectiv
                         resolution ??= result;
                     }
                     break;
+                case List<TElement> list:
```

Comment:
Just for my information, these blocks are duplicated only to have more efficient access to the elements when iteration, right? i.e. foreach over array being more efficient than foreach over `List<T>` etc.?

Reply:
Yes, more efficient/no alloc due to struct enumerator